006-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,95 @@
+
+The default behaviour of the `KafkaMirrorMaker2` operator and the `KafkaConnect` operator will be to deploy Kafka Connect clusters with a single TLS encrypted REST API listener.
+A self-signed TLS certificate and key required to enable TLS encryption on the REST API listener are generated by the operator, stored in secrets for use by the operator clients and the Kafka Connect pods.
+The certificates and keys are mounted in each Kafka Connect pod, which create an SSL keystore from the certificate and key and set the following properties in the generated connect configuration to enable an HTTPS listener:
+
+```
+listeners: https://:8443
+rest.advertised.listener: https
+rest.advertised.port: 8443
+listeners.https.ssl.client.auth: none
+listeners.https.ssl.keystore.location: /tmp/kafka/kafka-connect-rest.keystore.p12
+listeners.https.ssl.keystore.password: ***generated password***
+listeners.https.ssl.keystore.type: PKCS12
+```
+
+To communicate with the encrypted Connect REST API listener, the `KafkaMirrorMaker2`, `KafkaConnect` and `KafkaConnector` operators require modifications to the `AbstractConnectOperator` to apply the TLS certs as truststore options on the Vertx WebClient.
+
+### Enabling a plain HTTP REST API listener
+
+A new field is added to the `KafkaConnect` and `KafkaMirrorMaker2` specs named `restListeners`, which allows users to configure an additional unencrypted HTTP listener on port 8083 in the Kafka Connect configuration.
+This enables users to access the REST API without TLS if required.
+The `KafkaConnector`, `KafkaConnect` and `KafkaMirrorMaker2` operators will continue to use the TLS encrypted listener when an unencrypted HTTP listener is enabled.
+
+The `restListeners` field is designed for potential future extension, and is a list of items that contain a `protocol` field.
+For this proposal, the `protocol` field only supports the value `http`. One (or more) items in the `restListeners` list will add the unencrypted HTTP listener on port 8083.
+
+For example, the following CR will enable the HTTP listener on port 8083:
+```
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+spec:
+  restListeners:
+  - protocol: http
+  ...
+```
+
+The schema for `restListeners` is as follows:
+```
+openAPIV3Schema:
+  type: object
+  properties:
+    spec:
+      type: object
+      properties:
+        restListeners:
+          type: array
+          items:
+            type: object
+            properties:
+              protocol:
+                type: string
+                enum:
+                - http
+                description: The protocol of the REST listener.
+            required:
+            - protocol
+          description: List of additional REST listeners.
+        ...
+```
+
+#### Annotation alternative to enable a plain HTTP REST API listener
```

Comment:
So, is this a rejected alternative? If not, how / when are we going to decide on one or the other?

Reply:
I wasn't sure how folks would feel about the addition to the `spec`, so I added this as a possible alternative. Given that there haven't been objections (so far) to the `spec` addition, I have moved this annotation to the **Rejected Alternatives** section.